### PR TITLE
aggregate interference unit tests

### PR DESCRIPTION
**Pull request by janetliuboston**

### CODE REVIEW

**Comment by janetliuboston** [comment]
This PR is ready for review. Thanks!

File: src/harness/reference_models/examples/testutils.py [hunk 1]
```diff
@@ -80,3 +84,57 @@ def __call__(self,
...
+
+  Attributes:
+    ant_gain: Effective antenna gain from CBSD towards the protected entity
+              and vice versa. Default value is configured for interference
+              calculation
+    antenna_height: Antenna height considered for incidence angle calculation
+    reference_bw: Aggregate interference is calculated in reference bandwidth
+  """
+
+  def __init__(self,
```
Comment (sbdt-google) [inline src/harness/reference_models/examples/testutils.py]: Would have been nice to be able to pass a FakePropagationPredictor to the constructor, that is saved
in self.propag_predictor:
``` 
def __init__(self,
                  ant_gain=10,
                  reference_bw=5.e6,
                  propag_predictor=None):
      self.antenna_height = antenna_height
      self.ant_gain = ant_gain
      self.reference_bw = reference_bw
      if propag_predictor is None:
         propag_predictor = FakePropagationPredictor()
      self.propag_predictor = propag_predictor
```

File: src/harness/reference_models/examples/testutils.py [hunk 2]
```diff
@@ -80,3 +84,57 @@ def __call__(self,
...
+               reference_bw=5.e6):
+    self.antenna_height = antenna_height
+    self.ant_gain = ant_gain
+    self.reference_bw = reference_bw
+
+  def __call__(self, grant, eirp, constraint, fss_info=None,
+                   esc_antenna_info=None, region_type=None):
+
+    # Calculate the db loss and incidence angles using the FakePropagationPredictor
+    db_loss, incidence_angles, _ = FakePropagationPredictor().__call__(grant.latitude,
```
Comment (sbdt-google) [inline src/harness/reference_models/examples/testutils.py]: and then here:
```
db_loss, incidence_angles, _ = self.propag_predictor(grant.latitude, ...)
```

File: src/harness/reference_models/interference/aggregate_interference_test.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+
+  def setUp(self):
+    self.original_interference = interf.computeInterference
+
+  def tearDown(self):
+    interf.computeInterference = self.original_interference
+
+  def test_AggregateInterferenceFssCochannel(self):
+    expected_interference = {34.4116: {-100.57114: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0,\
```
Comment (sbdt-google) [inline src/harness/reference_models/interference/aggregate_interference_test.py]: Did you check by hand these values are expected, or are they the results of a first run ?

Of course if you pass a specific propag_predictor, with L1 distance (abs(lat2-lat1) + abs(lon2-lon1)), easy to know what is the expected pathloss returned, and then verify by hand/excel that the calculation is ok.

Same comment about below tests about making sure that the reference has been validated/computed by hand or in excel sheet, as it now tells you if there is an issue with the current code. The stored reference then provides regression testing for future update.

Comment (jni2000) [reply src/harness/reference_models/interference/aggregate_interference_test.py]: Have   changed the distance type to L1, calculated the path loss value manually and   verified the same with the executed results. Results are available up on request.

**Comment by jni2000** [comment]
Serge, we have addressed your review comments. Please take a look before we merge. Also, if test results are required, let us know where to save them.

**Comment by sbdt-google** [comment]
Thanks. Looks very good to me.
Since you have the test results, just add your excel  (or text file) with the results in the folder, as is.
You can call it  `aggregate_interference_test_calc_sheet.xlsx`  for example.

**Comment by jni2000** [comment]
Serge, added test results in " src/harness/reference_models/interference/test_results/", please take a look.

**Comment by sbdt-google** [comment]
I am fine, I havent checked the calculation detail as of now, but I trust the math works out ok.
Thanks, you can submit

**Pull request merged**